espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ViewFinder.java

```diff
@@ -197,11 +199,18 @@ public void describeTo(Description description) {
      * @param contentDesc Content description
      * @return
      */
-    private static boolean canScrollToViewWithContentDescription (String contentDesc) {
+    private static boolean canScrollToViewWithContentDescription (@Nullable final View parentView,
+                                                                  String contentDesc) {
         try {
             DataInteraction dataInteraction = onData(
                     hasEntry(Matchers.equalTo("contentDescription"), is(contentDesc))
             );
+
+            // If the parentView provided is an AdapterView, set 'inAdapterView' so that the
+            // selector is restricted to the adapter subtree
+            if (parentView != null && parentView instanceof AdapterView) {
```

Comment:
`parentView != null` is redundant since instanceof always returns false for null